linalg_ops/matmul/CMakeLists.txt

```diff
@@ -1,2036 +1,193 @@
+      iree-test-suites_iree-e2e-matmul-test
+    TARGET_BACKEND
+      "vmvx"
+    DRIVER
+      "local-task"
+    COMPILER_ARGS
+    RUNNER_ARGS
+    LABELS
+  )
+endforeach()
+
+###############################################################################
+#
+# GPU - Vulkan, default flags.
+#
+###############################################################################
+
+set(_DTYPES)
+# list(APPEND _DTYPES "i8_into_i32")  # Currently failing.
+list(APPEND _DTYPES "f32_into_f32")
+# list(APPEND _DTYPES "f16_into_f16")  # Failing to compile.
+# list(APPEND _DTYPES "f16_into_f32")  # Failing to compile.
+# list(APPEND _DTYPES "bf16_into_bf16")  # Failing to compile.
+# list(APPEND _DTYPES "bf16_into_f32")  # Failing to compile.
+# list(APPEND _DTYPES "f8E4M3FNUZ_into_f32")  # Unsupported data type.
+foreach(_DTYPE IN LISTS _DTYPES)
+  foreach(_SIZE IN LISTS _SIZES)
+    iree_test_suites_runner_test(
+      NAME
+        matmul_vulkan_${_DTYPE}_${_SIZE}
+      TESTS_SRC
+        "generated/${_DTYPE}/matmul_${_DTYPE}_${_SIZE}.mlir"
+      CALLS_SRC
+        "generated/${_DTYPE}/matmul_${_DTYPE}_${_SIZE}_calls.mlir"
+      TEST_RUNNER
+        iree-test-suites_iree-e2e-matmul-test
+      TARGET_BACKEND
+        "vulkan-spirv"
+      DRIVER
+        "vulkan"
+      COMPILER_ARGS
+      RUNNER_ARGS
+      LABELS
+    )
+  endforeach()
+endforeach()
+
+###############################################################################
+#
+# GPU - CUDA, default flags.
+#
+###############################################################################
+
+set(_DTYPES)
+list(APPEND _DTYPES "i8_into_i32")
+list(APPEND _DTYPES "f32_into_f32")
+# list(APPEND _DTYPES "f16_into_f16")  # Timeout running.
+list(APPEND _DTYPES "f16_into_f32")
+# list(APPEND _DTYPES "bf16_into_bf16")  # Timeout running.
+list(APPEND _DTYPES "bf16_into_f32")
+# list(APPEND _DTYPES "f8E4M3FNUZ_into_f32")  # Unsupported data type.
+foreach(_DTYPE IN LISTS _DTYPES)
+  foreach(_SIZE IN LISTS _SIZES)
+    iree_test_suites_runner_test(
+      NAME
+        matmul_cuda_${_DTYPE}_${_SIZE}
+      TESTS_SRC
+        "generated/${_DTYPE}/matmul_${_DTYPE}_${_SIZE}.mlir"
+      CALLS_SRC
+        "generated/${_DTYPE}/matmul_${_DTYPE}_${_SIZE}_calls.mlir"
+      TEST_RUNNER
+        iree-test-suites_iree-e2e-matmul-test
+      TARGET_BACKEND
+        "cuda"
+      DRIVER
+        "cuda"
+      COMPILER_ARGS
+      RUNNER_ARGS
+      LABELS
+    )
+  endforeach()
+endforeach()
+
+###############################################################################
+#
+# GPU - ROCm/HIP, default flags.
+#
+###############################################################################
+
+set(_DTYPES)
+list(APPEND _DTYPES "i8_into_i32")
+list(APPEND _DTYPES "f32_into_f32")
+list(APPEND _DTYPES "f16_into_f16")
+list(APPEND _DTYPES "f16_into_f32")
+list(APPEND _DTYPES "bf16_into_bf16")
+list(APPEND _DTYPES "bf16_into_f32")
+# list(APPEND _DTYPES "f8E4M3FNUZ_into_f32")  # Failing to compile.
```

Comment:
iree-org/iree tests ran tests on gfx94x/CDNA3 with the `LLVMGPUVectorDistributeMFMA` compilation info. Need to bake some target info into the test xpass/xfails.